benchkit/hdc/__init__.py

```diff
@@ -0,0 +1,219 @@
+# SPDX-License-Identifier: MIT
+"""
+Module to handle hdc (harmony debug bridge) interactions between host and remote phone.
+"""
+from typing import Iterable, Optional
+
+from benchkit.shell.shell import get_args, shell_out
+from benchkit.utils.types import Command, PathType
+
+
+def _identifier_from(ip_addr: str, port: int) -> str:
+    return f"{ip_addr}:{port}"
+
+
+class HDCError(Exception):
+    """Handle for errors from adb."""
+
+
+class HDCDevice:
+    """Representation of a device connected through hdc."""
+
+    def __init__(
+        self,
+        ip_addr: str,
+        port: str,
+    ) -> None:
+        self.ip_addr = ip_addr
+        self.port = port
+
+    def __str__(self) -> str:
+        return f"{self.ip_addr}:{self.port}"
+    
+
+class HarmonyDebugBridge:
+    """Operations with the phone for high-level hdc operations."""
+
+    def __init__(
+        self,
+        ip_addr: str,
```

Comment:
also here the function signature needs to be adapted, since the device identifier can be either a serial number or ip + port.